scan-engine/src/main/java/de/tillhub/scanengine/barcode/BarcodeScanner.kt

```diff
@@ -0,0 +1,10 @@
+package de.tillhub.scanengine.barcode
+
+import de.tillhub.scanengine.ScanEvent
+import kotlinx.coroutines.flow.Flow
+
+interface BarcodeScanner {
+    fun observeScannerResults(): Flow<ScanEvent>
+    fun scanWithKey(scanKey: String? = null)
+    fun unregisterBroadcastReceiver()
```

Comment:
Rename this method to something more ubiquitous, depending on later implementations we may not be working with broadcast receivers.

My suggestion would be "unregisterScannerListener" or something more fitting.